src/io/msh_vtkreader.cc

```diff
@@ -523,32 +518,22 @@ void VtkMeshReader::create_node_element_caches() {
 		return;
 	}
 
-	ASSERT( !has_compatible_mesh_ ).error();
-
 	has_compatible_mesh_ = true;
 
 	// read Points data section
-	HeaderQuery header_params("Points", 0.0, OutputTime::DiscreteSpace::MESH_DEFINITION);
-	auto point_header = this->find_header(header_params);
-	bulk_elements_id_.resize(point_header.n_entities);
-	for (unsigned int i=0; i<bulk_elements_id_.size(); ++i) bulk_elements_id_[i]=i;
-	this->get_element_data<double>(point_header.n_entities, point_header.n_components, false, 0 );
+	HeaderQuery points_query("Points", 0.0, OutputTime::DiscreteSpace::MESH_DEFINITION);
+	this->get_element_data<double>(points_query, false);
 
 	// read offset data section
-	HeaderQuery offsets_params("offsets", 0.0, OutputTime::DiscreteSpace::MESH_DEFINITION);
-    auto offset_header = this->find_header(offsets_params);
-    for (unsigned int i=bulk_elements_id_.size(); i<offset_header.n_entities; ++i) bulk_elements_id_.push_back(i);
-    std::vector<unsigned int> &offsets_vec = *( this->get_element_data<unsigned int>(offset_header.n_entities, offset_header.n_components, false, 0 ) );
+	HeaderQuery offsets_query("offsets", 0.0, OutputTime::DiscreteSpace::MESH_DEFINITION);
+    std::vector<unsigned int> &offsets_vec =
+        *( this->get_element_data<unsigned int>(offsets_query, false) );
 
     // read connectivity data section
-    HeaderQuery con_params("connectivity", 0.0, OutputTime::DiscreteSpace::MESH_DEFINITION);
-    auto & con_header = this->find_header(con_params);
-    con_header.n_entities = offsets_vec[offsets_vec.size()-1];
```

Comment:
after merging `find_header` and `get_element_data`, this abuse of the header is not possible and it causes problems..
not sure, how to overcome this.

Reply:
reverted